src/app/shared/store/compliance/compliance.reducer.ts

```diff
@@ -0,0 +1,63 @@
+import { createSelector } from 'reselect'
+
+import * as compliance from './compliance.actions'
+
+export interface State {
+  ids: string[]
+  entities: any
+  selectedId: string
+  isLoaded: boolean
+}
+
+const initialState: State = {
+  ids: [],
+  entities: [],
+  selectedId: '',
+  isLoaded: false
+}
+
+export function reducer (state = initialState, action: compliance.Actions): State {
+  switch (action.type) {
+
+    case compliance.GET_ALL: {
+      return Object.assign({}, state, {
+        isLoaded: false
+      })
+    }
+
+    case compliance.GET_ALL_SUCCESS: {
+      const payload = action.payload
+      const ids = payload
```

Comment:
Given the structure of this data and that it's always going to be completely updated there's no need to create `ids` and `entities`. Currently, they're redundant.

Suggestion: `data`